packages/bsky/src/api/app/bsky/feed/getPostThread.ts

```diff
@@ -1,97 +1,104 @@
 import { InvalidRequestError } from '@atproto/xrpc-server'
 import { Server } from '../../../../lexicon'
-import AppContext from '../../../../context'
-import {
-  FeedRow,
-  ActorInfoMap,
-  PostEmbedViews,
-  PostBlocksMap,
-} from '../../../../services/feed/types'
-import { FeedService, PostInfoMap } from '../../../../services/feed'
-import { Labels } from '../../../../services/label'
 import {
   BlockedPost,
   NotFoundPost,
   ThreadViewPost,
   isNotFoundPost,
 } from '../../../../lexicon/types/app/bsky/feed/defs'
+import { QueryParams } from '../../../../lexicon/types/app/bsky/feed/getPostThread'
+import AppContext from '../../../../context'
+import {
+  FeedService,
+  FeedRow,
+  FeedHydrationState,
+} from '../../../../services/feed'
 import {
   getAncestorsAndSelfQb,
   getDescendentsQb,
 } from '../../../../services/util/post'
 import { Database } from '../../../../db'
 import { setRepoRev } from '../../../util'
-
-export type PostThread = {
-  post: FeedRow
-  parent?: PostThread | ParentNotFoundError
-  replies?: PostThread[]
-}
+import { createPipeline, noRules } from '../../../../pipeline'
+import { ActorService } from '../../../../services/actor'
 
 export default function (server: Server, ctx: AppContext) {
+  const getPostThread = createPipeline(
+    skeleton,
+    hydration,
+    noRules,
```

Comment:
this should have rules right? around no blocks?

Reply:
Yeah, we could pretty easily yank 3p block-violating replies.  Blocks violating the viewer state are sort of part of the presentation, though, since they turn into `#blockedPost`s.

Reply:
ah right i see that 👍 

mer making me wish more and more that the response was flattened...

Reply:
maybe a comment here next to `noRules` saying that it's applied in presentation?

Reply:
Done!